### ISSUE TITLE
[calcite-icon] rename textLabel to intlIcon

### ISSUE THREAD

**Issue by jcfranco** [Esri/calcite-components#762]
This will make all localizable props consistent.

Stems from https://github.com/Esri/calcite-components/pull/592#discussion_r460250366

**Comment by jcfranco** [comment]
@driskull @paulcpederson I'm not sure what the new prop should be. Our pattern is `intl<DisplayedText>`, so should the icon prop be `intlIcon`? The main difference here is that there's no default value.

**Comment by driskull** [comment]
I think `intlLabel` would work fine. 

If we don't have a default maybe we should make it required. `!intlLabel: string;`. If we don't have anything provided the `aria-label` won't be added and it may not be as accessible as it should be.

**Comment by paulcpederson** [comment]
My understanding is that `int-` is for things that have an English default - like "close", provided by the component as a fallback. So in icon you may not need to update the prop name. We should decide and add it to our conventions doc.

**Comment by jcfranco** [comment]
I think icon is a special case since the label prop should only be used if the icon is semantic (see https://github.com/Esri/calcite-components/pull/293#issuecomment-581649522). Maybe we shouldn't treat it as a localizable prop and keep it as is?

**Comment by driskull** [comment]
Yeah. I guess this isn't really a `int` prop since it is unique per icon and is therefore more `content` than ui text.

**Comment by jcfranco** [comment]
Closing based on ☝️ .
